pyanaconda/modules/subscription/subscription.py

```diff
@@ -672,24 +673,9 @@ def unregister_with_task(self):
         # and clear attached subscriptions
         task.succeeded_signal.connect(
             lambda: self.set_attached_subscriptions([]))
```

Comment:
It might be easier to create a new callback and call all these functions from there.

Reply:
Yeah, that should reduce the boilerplate, lets do that.